ami/ml/orchestration/task_state.py

```
@@ -0,0 +1,126 @@
+        Args:
+            image_ids: List of image IDs that need to be processed
+            stages: List of stages to track for each image
+        """
+        self.stages = stages
+        for stage in stages:
+            cache.set(self._get_pending_key(stage), image_ids, timeout=self.TIMEOUT)
+
+        cache.set(self._total_key, len(image_ids), timeout=self.TIMEOUT)
+
+    def _get_pending_key(self, stage: str) -> str:
+        return f"{self._pending_key}:{stage}"  # noqa E231
```

Ruff (0.14.3)

Unused blanket `noqa` directive

Remove unused `noqa` directive

(RUF100)